witx/http-handler.witx

```diff
@@ -2,11 +2,41 @@
 ;;
 ;; The HTTP handler ABI defines functions invoked on an HTTP server handling
 ;; a client request.
+;;
+;; Note: This specification only uses numeric types, defined in WebAssembly
+;; Core Specification 1.0. For example, a string is passed as two parameters:
```

Comment:
Yeah this has always been an annoying part of witx

https://github.com/http-wasm/http-wasm-abi/pull/3#discussion_r977284500

Oh well, given we're not using codegen this seems better for now anyways

Reply:
maybe in the future we can alias the type as $string_ptr? I'll leave that call for when we have more functions, so it is easier to see.